### PR TITLE
Connect modal

### PR DESCRIPTION
**Pull request by utrennik**
Capture

*** Connect to lobby modal ***

In scope:

* create modal window (triggered by the start game button)
* create controlled form with fields:
- first name (required)
- last name (optional)
- avatar file upload (optional)
- avatar preview
- connect as obserwer switch
- confirm and cancel buttons (better to place confirm button on the right side, cancel - on the left side of the modal window)
* cancel button triggers modal close

Out of scope:
* confirm button click does nothing

Added name fields validation

### CODE REVIEW

File: client/src/components/pages/welcome-page/welcome-page.tsx [hunk 1]
```diff
@@ -1,47 +1,63 @@
...
+
+          <div className="connect-container error-down">
+            <div className="action-title">
+              Connect to lobby by <span>URL</span>:
+            </div>
+            <InputButton
+              buttonText="Connect"
+              valueHandler={(value: string) => {
+                handleOpen();
+                console.log(value);
```
Comment (aksiantsiuk) [inline client/src/components/pages/welcome-page/welcome-page.tsx]: Do you need `console.log`?

File: client/src/components/ui/customAvatar/customAvatar.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { ICustomAvatar } from '@models/types';
+
+export const CustomAvatar = ({ lastName, firstName, avatarImage }: ICustomAvatar) => {
+  const avatarName = lastName
+    ? `${firstName.match(/^./i)}${lastName.match(/^./i)}`.toUpperCase()
+    : `${firstName.match(/^./i)}${firstName.match(/.$/i)}`.toUpperCase();
+
+  return (
+    <Avatar
+      style={{ backgroundColor: '#60DABF', width: '50px', height: '50px' }}
```
Comment (aksiantsiuk) [inline client/src/components/ui/customAvatar/customAvatar.tsx]: replace `style`

File: client/src/models/types.ts [hunk 3]
```diff
@@ -16,3 +16,76 @@ export interface IInputButtonProps {
...
+  /*
+   * cancel button text
+   */
+  cancelBtnText: string;
+  /*
+   * modal close handler
+   */
+  onClose: () => void;
+
+  /*
```
Comment (aksiantsiuk) [inline client/src/models/types.ts]: Remove line break

File: client/src/components/pages/welcome-page/welcome-page.tsx (hunk 1 above)
Comment (utrennik) [reply client/src/components/pages/welcome-page/welcome-page.tsx]: Removed

File: client/src/models/types.ts (hunk 3 above)
Comment (utrennik) [reply client/src/models/types.ts]: Fixed

File: client/src/components/ui/customAvatar/customAvatar.tsx (hunk 2 above)
Comment (utrennik) [reply client/src/components/ui/customAvatar/customAvatar.tsx]: Fixed

File: client/src/components/modals/connect-modal/connect-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+          <CustomAvatar firstName={firstName || 'NA'} lastName={lastName} avatarImage={avatar} />
+        </div>
+
+        <div className="avatar-input">
+          <input
+            id="avatar-upload"
+            accept="image/*"
+            name="avatar-image"
+            type="file"
+            style={{ display: 'none' }}
```
Comment (aksiantsiuk) [inline client/src/components/modals/connect-modal/connect-modal.tsx]: className?

Comment (utrennik) [reply client/src/components/modals/connect-modal/connect-modal.tsx]: Fixed

**Review by aksiantsiuk — APPROVED** [review]

**Review by HelenBassa — APPROVED** [review]

**Review by RealOneReal — APPROVED** [review]

**Pull request merged**